nodeup/pkg/model/kubeapiserver.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package model
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/kops/pkg/flagbuilder"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/nodeup/nodetasks"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/v1"
+	metav1 "k8s.io/kubernetes/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/util/intstr"
+)
+
+// KubeAPIServerBuilder install kube-apiserver (just the manifest at the moment)
+type KubeAPIServerBuilder struct {
+	*NodeupModelContext
+}
+
+var _ fi.ModelBuilder = &KubeAPIServerBuilder{}
+
+func (b *KubeAPIServerBuilder) Build(c *fi.ModelBuilderContext) error {
+	if !b.IsMaster {
+		return nil
+	}
+
+	{
+		pod, err := b.buildPod()
+		if err != nil {
+			return fmt.Errorf("error building kube-apiserver pod: %v", err)
+		}
+
+		manifest, err := ToVersionedYaml(pod)
+		if err != nil {
+			return fmt.Errorf("error marshalling pod to yaml: %v", err)
+		}
+
+		t := &nodetasks.File{
+			Path:     "/etc/kubernetes/manifests/kube-apiserver.manifest",
+			Contents: fi.NewBytesResource(manifest),
+			Type:     nodetasks.FileType_File,
+		}
+		c.AddTask(t)
+	}
+
+	return nil
+}
+
+func (b *KubeAPIServerBuilder) buildPod() (*v1.Pod, error) {
+	flags, err := flagbuilder.BuildFlags(b.Cluster.Spec.KubeAPIServer)
+	if err != nil {
+		return nil, fmt.Errorf("error building kube-apiserver flags: %v", err)
+	}
+
+	redirectCommand := []string{
```

Comment:
Can you add a todo for setting this up to get logs from `kubectl logs` we have another issue in to allow for that.